### ISSUE TITLE
The BatchNorm layers are not completely frozen

### ISSUE THREAD

**Issue by jonbakerfish** [idstcv/SoftTriple#4]
The current implementation seems that the BatchNorm layers are not completely frozen.

The following lines only stop calculating the running mean & var, but the weight & bias of BN will still be changed:
https://github.com/idstcv/SoftTriple/blob/24173745e9c8d253c8e94a98690b94d8b18e6b4c/train.py#L138-L141

To avoid changing the weight & bias, the following lines:
https://github.com/idstcv/SoftTriple/blob/24173745e9c8d253c8e94a98690b94d8b18e6b4c/train.py#L82-L86
can be changed to:

```python
    if args.freeze_BN:
        for m in model.modules():
            if isinstance(m, torch.nn.BatchNorm2d):
                m.weight.requires_grad = False
                m.bias.requires_grad = False

    # define loss function (criterion) and optimizer
    criterion = loss.SoftTriple(args.la, args.gamma, args.tau, args.margin, args.dim, args.C, args.K).cuda()
    optimizer = torch.optim.Adam([{"params": filter(lambda p: p.requires_grad, model.parameters()), # model.parameters(), 
                                   "lr": args.modellr},
                                  {"params": criterion.parameters(), "lr": args.centerlr}],
                                 eps=args.eps, weight_decay=args.weight_decay)
```

**Comment by qian-qi** [comment]
Hi,
Yes, we only freeze the mean and var learned from ImageNet and compute the weights for the new data sets.
